File: app/src/main/java/org/scp/gymlog/util/LambdaUtils.java
package org.scp.gymlog.util;

import java.util.List;
import java.util.function.BiConsumer;
import java.util.function.Consumer;
import java.util.function.Predicate;

public class LambdaUtils {

    public static <T> Predicate<T> valueEquals(T compare) {
        return value -> value.equals(compare);
    }

    public static <I> void indexForEach(List<I> list, BiConsumer<Integer, I> consumer) {
        int index = 0;
        for (I element : list) {
            consumer.accept(index, element);
            index++;
        }
    }

    public static <I> void indexForEach(I[] list, BiConsumer<Integer, I> consumer) {
        int index = 0;
        for (I element : list) {
            consumer.accept(index, element);
            index++;
        }
    }
}

File: app/src/main/java/org/scp/gymlog/util/LambdaUtils.kt
package org.scp.gymlog.util

import java.util.function.BiConsumer
import java.util.function.Predicate

object LambdaUtils {
    fun <T> valueEquals(compare: T): Predicate<T> {
        return Predicate { value: T -> value == compare }
    }

    fun <I> indexForEach(list: List<I>, consumer: BiConsumer<Int, I>) {
        for ((index, element) in list.withIndex()) {
            consumer.accept(index, element)
        }
    }

    fun <I> indexForEach(list: Array<I>, consumer: BiConsumer<Int, I>) {
        for ((index, element) in list.withIndex()) {
            consumer.accept(index, element)
        }
    }
}
